### PR TITLE
Multilingual support

### PR DESCRIPTION
**Pull request by pmviva**
I completely added multilingual support in the theme.

A few things to note, Some of the strings and data were taken from data files. In order to make them translatable now you have one data file per language in a subfolder of the data directory. I took the idea from this hugo discussion:

https://discourse.gohugo.io/t/multilingual-data-files/4684

This allows to have customizable data per language

I updated the example site to reflect a real multilingual use case, having a title per language, some posts with two translations and multilingual data files

All the hard-coded strings are now externalized to a i18n file inside i18n folder, even the copyright messages

The languages in the menu are taken from the available translations of the current page displayed, so If you are viewing a page lets say "post 1" translated to english and spanish, you will have two menu entries, one for english and one for spanish, then if you move along to another post that only have 1 translation you won't see any entry in the menu

### CODE REVIEW

File: exampleSite/config-prod.toml [hunk 1]
```diff
@@ -1,5 +1,4 @@
 languageCode = "en-us"
-title = "Massively"
```
Comment (curttimson) [inline exampleSite/config-prod.toml]: I assume this is a breaking change?

All existing instances of the Massively theme will need to update their `config.toml` files?

Not a problem if so, just will need to update the version number.

Comment (pmviva) [reply exampleSite/config-prod.toml]: Well, don't know exactly what config-prod.toml in examplesSite is for, I assumed it was in order to generate the example hugo site to showcase the theme in hugo theme's page.

In multilingual mode you can override params by language, so you can have for example a title for English and a title for Spanish, if no param is found in the current language being generated it defaults to the global params.

Perhaps I should leave title as is and only override it for Spanish just to make it clear to people using the theme that you can override it.

**Comment by curttimson** [comment]
This looks really good! Would be great for the theme to have multilingual support.

Unfortunately I'm receiving this error when running locally:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x15994b4]

goroutine 1 [running]:
github.com/spf13/hugo/hugolib.(*Page).shouldBuild(0xc42044f200, 0xc4201dec00)
	/private/tmp/hugo-20170227-39762-14vq35b/hugo-0.19/src/github.com/spf13/hugo/hugolib/page.go:888 +0x34
github.com/spf13/hugo/hugolib.(*Site).convertSource(0xc420092280, 0x1768329)
	/private/tmp/hugo-20170227-39762-14vq35b/hugo-0.19/src/github.com/spf13/hugo/hugolib/site.go:1192 +0x227
github.com/spf13/hugo/hugolib.(*Site).createPages(0xc420092280, 0x175b415, 0x9)
	/private/tmp/hugo-20170227-39762-14vq35b/hugo-0.19/src/github.com/spf13/hugo/hugolib/site.go:1213 +0xc1
github.com/spf13/hugo/hugolib.(*Site).process(0xc420092280, 0x101, 0xc4201c18a0, 0xee1ed, 0xed2615c29)
	/private/tmp/hugo-20170227-39762-14vq35b/hugo-0.19/src/github.com/spf13/hugo/hugolib/site.go:820 +0xea
github.com/spf13/hugo/hugolib.(*HugoSites).process(0xc4201c4bd0, 0xc4201c1890, 0x0, 0x0, 0x0, 0x1b, 0x0)
	/private/tmp/hugo-20170227-39762-14vq35b/hugo-0.19/src/github.com/spf13/hugo/hugolib/hugo_sites_build.go:141 +0x126
github.com/spf13/hugo/hugolib.(*HugoSites).Build(0xc4201c4bd0, 0x101, 0x0, 0x0, 0x0, 0x0, 0xc42028a940, 0x1677d40)
	/private/tmp/hugo-20170227-39762-14vq35b/hugo-0.19/src/github.com/spf13/hugo/hugolib/hugo_sites_build.go:49 +0x102
github.com/spf13/hugo/commands.(*commandeer).buildSites(0xc42028e0e0, 0x1, 0x0, 0x1ad97f0)
	/private/tmp/hugo-20170227-39762-14vq35b/hugo-0.19/src/github.com/spf13/hugo/commands/hugo.go:721 +0xb5
github.com/spf13/hugo/commands.(*commandeer).build(0xc42028e0e0, 0xc420281afe, 0x1, 0x1, 0xc42028a240, 0x16)
	/private/tmp/hugo-20170227-39762-14vq35b/hugo-0.19/src/github.com/spf13/hugo/commands/hugo.go:499 +0xa4
github.com/spf13/hugo/commands.server(0x1add840, 0x1b014f8, 0x0, 0x0, 0x0, 0x0)
	/private/tmp/hugo-20170227-39762-14vq35b/hugo-0.19/src/github.com/spf13/hugo/commands/server.go:164 +0x849
github.com/spf13/hugo/vendor/github.com/spf13/cobra.(*Command).execute(0x1add840, 0x1b014f8, 0x0, 0x0, 0x1add840, 0x1b014f8)
	/private/tmp/hugo-20170227-39762-14vq35b/hugo-0.19/src/github.com/spf13/hugo/vendor/github.com/spf13/cobra/command.go:632 +0x3ef
github.com/spf13/hugo/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x1adbec0, 0x1788778, 0xc42006c058, 0x0)
	/private/tmp/hugo-20170227-39762-14vq35b/hugo-0.19/src/github.com/spf13/hugo/vendor/github.com/spf13/cobra/command.go:722 +0x339
github.com/spf13/hugo/commands.Execute()
	/private/tmp/hugo-20170227-39762-14vq35b/hugo-0.19/src/github.com/spf13/hugo/commands/hugo.go:167 +0x60
main.main()
	/private/tmp/hugo-20170227-39762-14vq35b/hugo-0.19/src/github.com/spf13/hugo/main.go:26 +0x32
```

Could this be a hugo version issue? It's currently set to support v0.19. If so, perhaps it's time to upgrade it's min version.

@pmviva Which version of Hugo are you running locally?

**Comment by curttimson** [comment]
@pmviva Thanks for checking, this will most definitely be a breaking change then.

Not a problem, but the `package.json` will need updating to reflect this.

File: exampleSite/config-prod.toml (hunk 1 above)
Comment (pmviva) [reply exampleSite/config-prod.toml]: @curttimson I reverted the change in config file regarding title, now it has a default value for English and is overriden in Spanish just to demonstrate how can be acheived.

**Pull request merged**